drivers: ieee802154_nrf5: add log for rx failure

lmaciejonczyk opened the pull request:
Some failures are possible and expected from fime to time e.g.
NRF_802154_RX_ERROR_TIMESLOT_ENDED. Add informational log for the frame
reception failure to differentiate the specific case.
It can be helpful for analizing failure in network trafic.

Signed-off-by: Lukasz Maciejonczyk <lukasz.maciejonczyk@nordicsemi.no>

lmaciejonczyk committed:
drivers: ieee802154_nrf5: add log for rx failure

Some failures are possible and expected from fime to time e.g.
NRF_802154_RX_ERROR_TIMESLOT_ENDED. Add informational log for the frame
reception failure to differentiate the specific case.
It can be helpful for analizing failure in network trafic.

Signed-off-by: Lukasz Maciejonczyk <lukasz.maciejonczyk@nordicsemi.no>

e-rk commented on drivers/ieee802154/ieee802154_nrf5.c:
```diff
@@ -956,6 +956,8 @@ void nrf_802154_receive_failed(nrf_802154_rx_error_t error, uint32_t id)
 		break;
 	}
 
+	LOG_INF("Rx failed, error = %d", error);
```
In a very noisy environment this log may printed very often. Is it really necessary to print this? Can it be a debug level log instead?

lmaciejonczyk replied:
The very noisy environment is where we would take the most benefit from the informational log here. If we change to debug log level we end up with many additional logs which will obfuscate the whole picture.

ankuns replied:
This is code is used by everyone. This logging could be under some `#if`. If you need so deep logging then you can enable this option on your own.
Possibly:
```c
#if defined(CONFIG_IEEE802154_NRF5_LOG_RX_FAILURES)
   LOG_INF("Rx failed, error = %d", error);
#endif
```
and `config IEEE802154_NRF5_LOG_RX_FAILURES` option turned off by default in nearby Kconfig.nrf5

lmaciejonczyk replied:
We can set log level for this specific module/file so you can still disable logs if you are not interested in. There is already Kconfig for that: `CONFIG_IEEE802154_DRIVER_LOG_LEVEL` so we don't need to add extra one.
What I find most interest among all logs in this module is this what I want to add here but combining it with others with the debug log level has not much sense to me as it will not be much help in that shape.

ankuns replied:
@lmaciejonczyk  yes, you do. By doing so you will have the feature you need, other users of shared shim will not be affected.

e-rk reviewed (changes requested):
I have one concern stated above.

rlubos commented on drivers/ieee802154/ieee802154_nrf5.c:
```diff
@@ -956,6 +956,10 @@ void nrf_802154_receive_failed(nrf_802154_rx_error_t error, uint32_t id)
 		break;
 	}
 
+#if IS_ENABLED(CONFIG_IEEE802154_NRF5_LOG_RX_FAILURES)
```
Would be good to use regular `if (IS_ENABLED(CONFIG_IEEE802154_NRF5_LOG_RX_FAILURES)) {`, so that the code is still compiled.

lmaciejonczyk committed:
drivers: ieee802154_nrf5: fix rx failure error propagation for CSL

For delayed reception only NRF_802154_RX_ERROR_DELAYED_TIMEOUT is
expected to happen, others rx errors should be handled in regular
manner.

Signed-off-by: Lukasz Maciejonczyk <lukasz.maciejonczyk@nordicsemi.no>

lmaciejonczyk commented:
I was force to rebase the PR to resolve merge problems.
